### PR TITLE
Update Return Value To $status

### PR DESCRIPTION
**Pull request by mycraftmw**
I updated the return value to `$status`, and I add a comment for it to avoid some stupid errors in the future.

### CODE REVIEW

File: functions/__z.fish [hunk 1]
```diff
@@ -164,9 +164,14 @@ function __z -d "Jump to a recent directory."
...
-            type -q xdg-open;and xdg-open "$target"; and return $status;
-            type -q open;and open "$target"; and return $status;
-            echo "Not sure how to open file manager"; and return 1;
+            if test "$OS" = Windows_NT
+                type -q explorer;and explorer "$target"; return $status;
+                # Be careful, in msys2, explorer always return 1
+            else
+                type -q xdg-open;and xdg-open "$target"; and return $status;
+                type -q open;and open "$target"; and return $status;
+                echo "Not sure how to open file manager"; and return 1;
```
Comment (jethrokuan) [inline functions/__z.fish]: Let's move this line outside the if statement, so it also covers the case where `explorer` doesn't exist (although highly unlikely).

File: functions/__z.fish [hunk 2]
```diff
@@ -164,9 +164,14 @@ function __z -d "Jump to a recent directory."
         if set -q _flag_echo
             printf "%s\n" "$target"
         else if set -q _flag_directory
-            type -q xdg-open;and xdg-open "$target"; and return $status;
-            type -q open;and open "$target"; and return $status;
-            echo "Not sure how to open file manager"; and return 1;
+            if test "$OS" = Windows_NT
+                type -q explorer;and explorer "$target"; return $status;
+                # Be careful, in msys2, explorer always return 1
```
Comment (krobelus) [inline functions/__z.fish]: I think @jethrokuan [said](https://github.com/jethrokuan/z/pull/76/files#r263267402) that we should return 0 here.

Comment (mycraftmw) [reply functions/__z.fish]: Oh, @jethrokuan, so which one do you expect?

Comment (jethrokuan) [reply functions/__z.fish]: let's keep it at 0 to be standard.

Comment (mycraftmw) [reply functions/__z.fish]: OK.

File: functions/__z.fish (hunk 1 above)
Comment (mycraftmw) [reply functions/__z.fish]: It's done now.

Comment (jethrokuan) [reply functions/__z.fish]: Apologies if I wasn't clear, what i meant was:

if `type -q explorer` fails, the `zo` command will currently do nothing at all, whereas I expect it to output `"Not sure how to open file manager"` and return 1.

This fallback is currently implemented only for unix, but we should make that fallback cover windows as well, so moving the fallback out of the if block would achieve that behaviour.

**Comment by jethrokuan** [comment]
lgtm, thanks! :+1:

**Comment by mycraftmw** [comment]
My pleasure! 😁

**Pull request merged**